### PR TITLE
Mainline hobo stove and tinder from More Survival Tools

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary
 ```SUMMARY: Content "Adds MST hobo stove, adjusts usage to reheating/defrosting only."```

#### Purpose of change
Mainline the hobo stove and balance it for it's size.  

#### Describe the solution

It will reheat and defrost food & water and uses 25 charges per action.  I haven't changed the tinder or overall volume of the hobo stove.  So that's 2 actions per load of tinder.

Also limited its recipe to food cans only, since soft drink cans are softer metal.

#### Describe alternatives you've considered

keeping boiling quality.

#### Additional context
action shot!
hobo stove

### CODE REVIEW

**Comment by Scischm** [comment]
The stove itself is 1.5 liters and 100 tinder is .5 liters. If use per action is increased to 25 per I would say that upping the amount of tinder that can be stored in the stove isn't a bad idea. 1/3 of the cans volume at 100 charges would make sense for a dedicated fuel area. 

Wouldn't hurt overall, just reduce some keypresses since the reload menu sometimes likes to be wonky and prevent you from just dumping tinder straight into the hobo stove. (Sometimes have to close off crafting menu or heating menu, open inventory, select the stove, hit r, reopen the crafting/heating menu, find the item you were working on before, etc.)

**Comment by Throwaway-name** [comment]
I'd keep the hotplate functionality. A cursory look at the real ones shows they can cook soup and tacos with a pan.

**Comment by anothersimulacrum** [comment]
The real ones are also much larger than this one is. If you look up hobo stove, you don't see with a tin can the size they are in game. If you wanted to have a hobo stove with hotplate functionality, the first step would be to add larger cans.

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -1159,5 +1159,27 @@
...
+    "bashing": 1,
+    "material": "wood",
+    "flags": [ "NO_SALVAGE" ],
+    "symbol": "=",
+    "color": "brown",
+    "count": 50,
+    "ammo_type": "tinder",
+    "range": 0,
+    "damage": 0,
+    "dispersion": 0,
```
Comment (Night-Pryanik) [inline data/json/items/ammo.json]: Range, damage and dispersion fields are optional and could be removed.

File: data/json/items/tools.json [hunk 2]
```diff
@@ -8762,5 +8762,65 @@
...
+    "color": "light_gray",
+    "ammo": "tinder",
+    "max_charges": 50,
+    "charges_per_use": 25,
+    "use_action": [
+      {
+        "target": "hobo_stove_on",
+        "msg": "The hobo stove is lit.",
+        "need_fire": 1,
+        "need_fire_msg": "You need a lighter!",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]:
```suggestion
        "need_fire_msg": "You need a source of fire!",
```

**Comment by Night-Pryanik** [comment]
Difficulty 4 seems to be rather high for a 5-minutes-to-craft thing made out of metal can and a metal scrap.

**Comment by Rail-Runner** [comment]
> Also limited its recipe to food cans only, since soft drink cans are softer metal.

Would material softness matter considering how it's supposed to be used? I'd also consider adding empty metal canisters (or maybe even clay canisters) as alternative to tin cans, so that one could craft such a thing from scratch if desired.

**Comment by curstwist** [comment]
> 
> 
> I'd keep the hotplate functionality. A cursory look at the real ones shows they can cook soup and tacos with a pan.

as anothersimulacrum mentioned, there's been talk of a larger hobo stove to add cooking options back in.  Something like 30oz coffee cans vs. the 10.5oz cans for vegetables/soups which is what is represented in this recipe.

**Comment by curstwist** [comment]
> 
> 
> > Also limited its recipe to food cans only, since soft drink cans are softer metal.
> 
> Would material softness matter considering how it's supposed to be used? I'd also consider adding empty metal canisters (or maybe even clay canisters) as alternative to tin cans, so that one could craft such a thing from scratch if desired.

I'd like to see where the discussion on new canister sizes goes before adding some (which also means adding some other content for them).  You can use some scrap to make this instead of a can, just not soda cans.

**Comment by Scischm** [comment]
There are also bulk food cans that would probably work nicely for a hobo stove. I.e. 
image

**Comment by curstwist** [comment]
jenkins rebuild

**Comment by curstwist** [comment]
please notify me if this is waiting on me to change something.

**Pull request merged**